### PR TITLE
Code Quality fixes 

### PR DESCRIPTION
**Pull request by lakshyacis**
# Changes based on Flake8 and Pylint fixes.
announcethread.py Flake8 fixes.
announcethread.py Pylint fixes.
bmproto.py Pylint fixes.
connectionpool.py Pylint fixes.
dandelion.py Pylint fixes.
downloadthread.py Flake8 fixes.
downloadthread.py Pylint fixes.

### CODE REVIEW

File: src/network/announcethread.py [hunk 1]
```diff
@@ -10,6 +14,7 @@
 
 
 class AnnounceThread(StoppableThread):
+    """Method while Announce the thread."""
```
Comment (g1itch) [inline src/network/announcethread.py]: Better use docstrings from #1396

Comment (lakshyacis) [reply src/network/announcethread.py]: @g1itch sure!

File: src/network/announcethread.py [hunk 2]
```diff
@@ -25,9 +30,14 @@ def run(self):
                 self.stop.wait(10)
 
     def announceSelf(self):
+        """Announce our presence"""
+        # pylint: disable=no-self-use
```
Comment (g1itch) [inline src/network/announcethread.py]: Maybe `@staticmethod` was the right decision, why did you change?

Comment (lakshyacis) [reply src/network/announcethread.py]: Might be, it was mistakenly done.

File: src/network/connectionpool.py [hunk 3]
```diff
@@ -113,7 +118,7 @@ def removeConnection(self, connection):
                     pass
         connection.handle_close()
 
-    def getListeningIP(self):
+    def getListeningIP(self):       # pylint: disable=no-self-use
```
Comment (PeterSurda) [inline src/network/connectionpool.py]: this can probably be made into a `@staticmethod`

File: src/network/connectionpool.py [hunk 4]
```diff
@@ -253,8 +259,8 @@ def loop(self):
                     self.startListening()
                 else:
                     for bind in re.sub(
-                        "[^\w.]+", " ",
-                        BMConfigParser().safeGet('network', 'bind')
+                            "[^\w.]+", " ",     # pylint: disable=anomalous-backslash-in-string
```
Comment (PeterSurda) [inline src/network/connectionpool.py]: maybe this should be a `'` instead of `"`?

File: src/network/connectionpool.py [hunk 5]
```diff
@@ -263,8 +269,8 @@ def loop(self):
                     self.startUDPSocket()
                 else:
                     for bind in re.sub(
-                        "[^\w.]+", " ",
-                        BMConfigParser().safeGet('network', 'bind')
+                            "[^\w.]+", " ",     # pylint: disable=anomalous-backslash-in-string
```
Comment (PeterSurda) [inline src/network/connectionpool.py]: Same here.

File: src/network/connectionpool.py (hunk 3 above)
Comment (lakshyacis) [reply src/network/connectionpool.py]: Sometimes, @staticmethod breaks the code .So, that's why I disable the warning. 
But, will update and correct the code. :)

File: src/network/connectionpool.py (hunk 4 above)
Comment (lakshyacis) [reply src/network/connectionpool.py]: Okay, will update the code.

File: src/network/connectionpool.py (hunk 5 above)
Comment (lakshyacis) [reply src/network/connectionpool.py]: Okay

File: src/network/connectionpool.py (hunk 3 above)
Comment (PeterSurda) [reply src/network/connectionpool.py]: Yes, sometimes it breaks, but I looked at it an in this case it's ok.

File: src/network/bmproto.py [hunk 6]
```diff
@@ -422,16 +430,16 @@ def _decode_addr(self):
...
             decodedIP = protocol.checkIPAddress(str(ip))
             if stream not in state.streamsInWhichIAmParticipating:
                 continue
             if (
-                decodedIP and time.time() - seenTime > 0 and
-                seenTime > time.time() - BMProto.addressAlive and
-                port > 0
+                    decodedIP and time.time() - seenTime > 0 and
+                    seenTime > time.time() - BMProto.addressAlive and
+                    port > 0
```
Comment (g1itch) [inline src/network/bmproto.py]: I was trying to save horizontal space and this should not violate PEP8 or any other style guide...

Comment (g1itch) [reply src/network/bmproto.py]: So why 996e71a was merged without any warning?

Comment (lakshyacis) [reply src/network/bmproto.py]: I think github check didn't run correctly.

Comment (PeterSurda) [reply src/network/bmproto.py]: I think it may be a bug with the checks. I'm getting the same error locally.

Comment (g1itch) [reply src/network/bmproto.py]: This is really a bug: https://github.com/PyCQA/pylint/issues/289

Comment (PeterSurda) [reply src/network/bmproto.py]: For the checks GitHub App, pylint 1.6.5 is used, whereas locally I have 1.9.2. I don't want to change anything as the server seems to misbehave and there is no failover. We'll have to wait until I can put the new system using jenkins into production. I hope it will be by the end of this month.

Comment (PeterSurda) [reply src/network/bmproto.py]: I would say we leave these pylint problems for later after the checks are upgraded.

Comment (g1itch) [reply src/network/bmproto.py]: I meant we could do something like
```diff
diff --git a/setup.cfg b/setup.cfg
index 3236eed1..68a8181c 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -4,6 +4,10 @@
 [pycodestyle]
 max-line-length = 119
 
+[pylint.messages control]
+disable = C0330
+# C0330 - wrong-hanging-indentation
+
 [flake8]
 max-line-length = 119
 ignore = E722,F841
```

and invoke pylint with `--rcfile setup.cfg` because if you are going to format the code according to pylint bugs it will become unreadable soon.
